File: 1-primeros-algoritmos/4-pow/pow.cpp
#include<iostream>

using namespace std;

int pow(double base, int exp){
    if(exp == 0)
        return 1;
    return base * pow(base, exp-1);
}

int main(){

    cout << "base = ";
    int base;
    cin >> base;
    cout << endl;

    cout << "exp = ";
    int exp;
    cin >> exp;
    cout << endl;

    cout << base << " elevado a la " << exp << " es igual a: " << pow(base, exp) << endl;
    
    return 0;
}

File: 2-divide-y-conquista/1-pow/pow.py


def pow(base, exp):
    if exp == 0:
        return 1
    
    if exp % 2 == 0:
        aux = pow(base, exp / 2)
        return aux * aux
    
    return base * pow(base, exp - 1)


print(pow(5, 3))